python/lsst/scarlet/lite/observation.py

```diff
@@ -366,6 +367,29 @@ def __getitem__(self, indices: Any) -> Observation:
         new_variance = self.variance[indices]
         new_weights = self.weights[indices]
 
+        # If the indices is a single band, make sure to keep the band axis
+        if new_image.ndim == 2:
+            if indices in self.bands:
```

Comment:
It's not totally obvious why this implies indices is a band (which is always a str?). Can't/shouldn't you check that explicitly?